pkg/controller/controller.go

```diff
@@ -489,7 +489,7 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 			// DataSource is not VolumeSnapshot and PVC
 			// Wait for an external data populator to create the volume
 			p.eventRecorder.Event(options.PVC, v1.EventTypeNormal, "Provisioning", fmt.Sprintf("Waiting for a volume to be created by an external data populator"))
-			return nil, controller.ProvisioningFinished, nil
+			return nil, controller.ProvisioningFinished, fmt.Errorf("data source %s not handled by external provisioner", options.PVC.Spec.DataSource.Kind)
```

Comment:
Failed `Provision` call emits an event and at the same time there is event "Waiting for a volume to be created by an external data populator" just above. Won't they send confusing message to the user?

Reply:
Yes, you do get a bunch of confusing events in that case. I want to clean up the event situation as part of my populator work, but this fix is just to prevent the panic. If you want I can additionally remove the preceding line to prevent that event. I'm not sure who added that event.

Reply:
> What appears to be going on is that that when ProvisionController.provisionClaimOperation() calls csiProvisioner.Provision(), the callee is returning a nil volume and a nil error. That caller assumes that nil error implies that volume is non-nil and tries to dereference it for logging purposes.

This returns "nil" here so that the volume can be handled by an external data populator. The logic in the caller should be fixed and not to assume that volume is non-nil if error is nil here.

Reply:
An alternative would be to return a specific error code here.  The caller will check the error code and recognizes this failure can be ignored and will be handled by an external data populator.